Task queue celery

gaurdianaq opened the pull request:
Added a bunch of refactoring, task queues now actually work...

Commit:
Removed all references to flask and the database in the reddit crawler

Commit:
Merge branch 'dev' of https://github.com/hatchways/team-ginger into task_queue

Commit:
Pulled in request module, updated crawlers to use a uniform class type for storing data

Commit:
added data response, reorded responses by response code to make reading easier

Commit:
Enqueue written, responses route written, mention class that's not related to the db made, the moment of reckoning approaches, it's almost time to test...

Commit:
requests route is ready to be tested

Commit:
backend part of this is done, time to update it on the front end

Commit:
Temporarily giving up on rq

Commit:
moved a bunch of stuff around for task queues

Commit:
Changes still seem to be failing

Commit:
started implemented celery but need to go back and test something

Commit:
So far having more luck with celery

Commit:
now actually runs the task! Need to make it repeat

Commit:
queues are working

Commit:
Updated read me

Commit:
Removed site_id from mention class as it's repetitive and doesn't need to be included so many times in the json request sent by the task

Commit:
Removed user_id from mention class as again, redundant

Commit:
Moved constants that are shared between the crawlers and the flask app to a new file

Commit:
Refactored a bunch of json tags to use constants

Commit:
Repeated tasks are working, just need to be able to remove them and get rid of duplicates

Commit:
Added flower monitor, task queues now seem to be working for real!

Commit:
merge conflicts fixed, I think

ryanarine commented on README.md:
```diff
@@ -4,5 +4,8 @@
-3. Activate the virtual environment and start the app with `pipenv run flask run`
+2. Install the dependencies with `pipenv install`. This also create a a virtual environment, if there isn't one already
+3. Activate the virtual environment `pipenv shell`
+4. Ensure postgres & redis are installed
```
This should be before step 3

Commit:
merged in sentiment analysis branch, fixed merge conflicts

Commit:
Added pipfile.lock back

ryanarine commented on server/mentions_crawler_flask/responses/__init__.py:
```diff
@@ -8,55 +9,65 @@
 _COMPANY_NAMES_TAG = "names"
```
This should be equal to "companies" because you changed the corresponding frontend constant to that

ryanarine reviewed (changes requested):
Small bug due to changing one of the constants but otherwise it looks like its working. I'll fix it and merge.

ryanarine committed:
Constant Tag Fix
-Fixed company names tag constant

ryanarine committed:
I(Ryan) Forgot to use tag

Fixed another bug that was caused due to the company name tag change. I forgot to use the constant tag.
